elastic/kibana
# [Security Solution][Detections] Add alert source to detection rule action context

## Summary
This PR adds the source of the detection alert documents to the rule action context so notification actions can access fields within the detection alerts. The source alerts are available as an array through Mustache templating at `{{context.alerts}}`, e.g. `{{context.alerts.0.process.name}}` would get the process.name field of the first alert.

Docs update: Alert data is now available in detection rule actions at {{context.alerts}} as an array. This array contains each alert generated since the last time the action executed. Mustache templating can be used to iterate over all alerts in the array and capture information from each one. For example, `{{#context.alerts}}Detection alert for user: {{user.name}}{{/context.alerts}}` would create the string `Detection alert for user: <user.name>` for every alert in the array. Any alerts that don't have `user.name` will still generate the string but leave `<user.name>` blank.

![image](https://user-images.githubusercontent.com/616158/103748744-9109d800-4fd2-11eb-9d36-3b8accdf8cce.png)

Test process:
1. Go through the rule creation process for a rule type up to step 4 (Rule Actions).
2. Create a webhook connector that POSTs to `http://localhost:<some port>` with no auth and no http header, ex
![image](https://user-images.githubusercontent.com/55718608/101698040-dd4f1000-3a46-11eb-9112-1c73561191af.png)
3. Choose either "On rule execution" or one of the throttled options for when the action should fire. If using the throttled options, change the throttle time in `/kibana/x-pack/plugins/security_solution/public/detections/components/rules/throttle_select_field/index.tsx` down to 5m or less so you don't have to wait an hour for the action to trigger
4. Use netcat or equivalent to listen on the chosen port. e.g. `sudo nc -l 392` on ubuntu listens on port 392.
5. Create a document that triggers the rule. When the action fires, netcat should print out the request it received from the webhook. You may need to restart netcat for it to print further requests even though the process continues to run.

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
